core/tx_pool.go

```diff
@@ -1681,6 +1723,33 @@ func (pool *TxPool) demoteUnexecutables() {
 	}
 }
 
+// insertTxToLifecycle insert the tx to lifecycle calculator
+func (pool *TxPool) insertTxToLifecycle(tx common.Hash) {
+	if !pool.enableTxLifecycles {
+		return
+	}
+
+	pool.txLifecycles.Add(tx, uint64(time.Now().Unix()))
```

Comment:
Do we need to check if the transaction is already in `txLifecycles` (e.g. resubmission)?

Reply:
LRU cache will check this